kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/BaseOperation.java

```diff
@@ -291,31 +290,27 @@ public R load(String path) {
   @Override
   public final T createOrReplace(T... items) {
     T itemToCreateOrReplace = getItem();
+    Resource<T> resource;
     if (items.length > 1) {
       throw new IllegalArgumentException("Too many items to create.");
     } else if (items.length == 1) {
       itemToCreateOrReplace = items[0];
+      resource = withItem(itemToCreateOrReplace);
+    } else {
+      resource = this;
     }
```

Comment:
I find this sort of confusing, maybe replacing with the following might help:
```suggestion
    final T itemToCreateOrReplace;
    Resource<T> resource;
    if (items.length > 1) {
      throw new IllegalArgumentException("Too many items to create.");
    } else if (items.length == 1) {
      itemToCreateOrReplace = items[0];
      resource = withItem(itemToCreateOrReplace);
    } else {
      itemToCreateOrReplace = getItem();
      resource = this;
    }
```